src/overlays/actors/ovl_En_Bubble/z_en_bubble.c

```diff
@@ -32,73 +31,399 @@ const ActorInit En_Bubble_InitVars = {
+        this->velocityFromBounce.x = (this->bounceDirection.x * bounceSpeed);
+        this->velocityFromBounce.y = (this->bounceDirection.y * bounceSpeed);
+        this->velocityFromBounce.z = (this->bounceDirection.z * bounceSpeed);
+        this->sinkSpeed = 0.0f;
+        Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_AWA_BOUND);
+        this->graphicRotSpeed = 128.0f;
+        this->graphicEccentricity = 0.48f;
+    }
+    this->actor.velocity.x = this->velocityFromBounce.x + this->velocityFromBump.x;
+    this->actor.velocity.y = this->velocityFromBounce.y + this->velocityFromBump.y + this->sinkSpeed;
+    this->actor.velocity.z = this->velocityFromBounce.z + this->velocityFromBump.z;
+    Math_ApproachF(&this->velocityFromBump.x, 0.0f, 0.3f, 0.1f);
+    Math_ApproachF(&this->velocityFromBump.y, 0.0f, 0.3f, 0.1f);
+    Math_ApproachF(&this->velocityFromBump.z, 0.0f, 0.3f, 0.1f);
+}
+
+u32 func_8089FF30(EnBubble* this) {
+    if (((this->colliderSphere.base.acFlags & AC_HIT) != 0) == false) {
+        return false;
+    }
+    this->colliderSphere.base.acFlags &= ~AC_HIT;
+    if (this->colliderSphere.elements[1].info.bumperFlags & BUMP_HIT) {
+        this->unk1F4.x = this->colliderSphere.base.ac->velocity.x / 10.0f;
+        this->unk1F4.y = this->colliderSphere.base.ac->velocity.y / 10.0f;
+        this->unk1F4.z = this->colliderSphere.base.ac->velocity.z / 10.0f;
+        this->graphicRotSpeed = 128.0f;
+        this->graphicEccentricity = 0.48f;
+        return false;
+    }
+    this->unk200 = 8;
+    return true;
+}
+
+s32 EnBubble_DetectPop(EnBubble* this, PlayState* play) {
+    if (DECR(this->unk200) != 0 || this->actionFunc == EnBubble_Pop) {
+        return false;
+    }
+    if (this->colliderSphere.base.ocFlags2 & OC2_HIT_PLAYER) {
+        this->colliderSphere.base.ocFlags2 &= ~OC2_HIT_PLAYER;
+        EnBubble_DamagePlayer(this, play);
+        this->unk200 = 8;
+        return true;
+    }
+    return func_8089FF30(this);
+}
+
+void func_808A005C(EnBubble* this) {
+    ColliderJntSphElementDim* dim;
+    Vec3f src;
+    Vec3f dest;
+
+    dim = &this->colliderSphere.elements[0].dim;
+    src.x = dim->modelSphere.center.x;
+    src.y = dim->modelSphere.center.y;
+    src.z = dim->modelSphere.center.z;
+
+    Matrix_MultVec3f(&src, &dest);
+    dim->worldSphere.center.x = dest.x;
+    dim->worldSphere.center.y = dest.y;
+    dim->worldSphere.center.z = dest.z;
+    dim->worldSphere.radius = dim->modelSphere.radius * (1.0f + this->expansionWidth);
+    this->colliderSphere.elements[1].dim = *dim;
+}
+
+void EnBubble_Init(Actor* thisx, PlayState* play) {
+    EnBubble* this = THIS;
+    s32 pad;
+
+    ActorShape_Init(&this->actor.shape, 16.0f, ActorShadow_DrawCircle, 0.2f);
+    Collider_InitJntSph(play, &this->colliderSphere);
+    Collider_SetJntSph(play, &this->colliderSphere, &this->actor, &sJntSphInit, this->colliderElements);
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, DamageTable_Get(9), &sColChkInfoInit);
```

Comment:
Oh neat, first time I've seen a generic damagetable that wasn't used for a dialogue NPC, was starting to wonder if they actually used it for a damageable enemy.

(except this actor is unused in the final game ...)